### PR TITLE
Quests/npc

### LINKED ISSUE #1558: Quests: Write functions that help NPC dialogue determine if the task/quest can be given
We want to have a conditional NPC dialogue that will only be triggered if the player meets the necessary prereqs for the particular quest/task. Instead of having the NPC modules depend on prereqs, we will only need to pass ids and a qctx struct. After these functions are complete, the NPC modules can call on them to start the dialogue

### PR DESCRIPTION
**Pull request by elizabethli31**
We wrote two functions to help the NPC dialogue know whether to initiate convo about giving a task or quest depending on the prerequisites that the player has and if it meets the prerequisites of the task/quest. We wanted to write functions for the NPC team, so that they do not need to rely on (or initialize) our structs. The only parameters that the NPC team needs in their module is the ids (which is part of a different struct that we wrote for the NPC struct) and quest hash tables which hold all the quests in a given game. `npc_can_give_quest` and `npc_can_give_task` both return a bool which NPC can use to see if they should start the conversation.

### CODE REVIEW

File: src/quests/src/quests_state.c [hunk 1]
```diff
@@ -285,3 +285,39 @@ int update_player_quests(quest_ctx_t *qctx) {
...
+{
+    assert(qctx != NULL);
+    quest_t *quest = get_quest_from_hash(quest_id, qctx->quest_hash);
+
+    assert(quest != NULL);
+    
+    prereq_t *prereq = quest->prereq;
+    player_t *player = qctx->player;
+
+    if (meets_prereqs(player, prereq)) {
```
Comment (JSerf02) [inline src/quests/src/quests_state.c]: Can be shortened to just be return meets_prereqs(player, prereq)

File: src/quests/src/quests_state.c [hunk 2]
```diff
@@ -285,3 +285,39 @@ int update_player_quests(quest_ctx_t *qctx) {
...
+{
+    assert(qctx != NULL);
+    task_t *task = get_task_from_quest_hash(task_id, qctx->quest_hash);
+
+    assert(task != NULL);
+    
+    prereq_t *prereq = task->prereq;
+    player_t *player = qctx->player;
+
+    if (meets_prereqs(player, prereq)) {
```
Comment (JSerf02) [inline src/quests/src/quests_state.c]: Can be shortened to just be return meets_prereqs(player, prereq)

**Review by JSerf02 — APPROVED** [review]
Changes look great! A commented 2 areas of code that can be shortened for better readability and less redundancy, but otherwise, everything seems good.

**Review by philipchenaux — APPROVED** [review]
Nice set of functions, looks good to merge!

**Comment by philipchenaux** [comment]
Issue Score: Excellent

Comments:
Excellent job on this PR!

**Pull request merged**